packages/firestore/src/model/field_value.ts

```diff
@@ -605,6 +606,31 @@ export class ObjectValue extends FieldValue {
     return field;
   }
 
+  /** Recursively extracts the FieldPaths that are set in this ObjectValue. */
```

Comment:
I feel like the comments here can be elaborated, something like:

Returns a `FieldMask` built from all `FieldPath`s starting from this `ObjectValue`, including paths from nested objects.

Reply:
Done (went with your version).